lib/IRGen/GenProto.cpp

```diff
@@ -384,6 +390,10 @@ void PolymorphicConvention::considerParameter(SILParameterInfo param,
     case ParameterConvention::Direct_Owned:
     case ParameterConvention::Direct_Unowned:
     case ParameterConvention::Direct_Guaranteed:
+      // Don't consider ClassPointer source and Metadata source for partial_apply forwader with non fixed layout.
+      // If not, we may end up with missing TypeMetadata for a type dependent generic parameter
+      // while generating code for destructor of HeapLayout.
+      if (!ConsiderParameterSources) return;
```

Comment:
Is the check here still neccessary if you add a check:  
```
  if (ConsiderParameterSources)
   considerParameter(params[selfIndex], selfIndex, true);
```

to line 221?

Reply:
Forgot to clean it up. Will fix in the following commit.